### PR TITLE
improve err handling, use generalized err message

### LINKED ISSUE #302: Home page may display errors that are meant for internal server use
The home page should only display messages intended to be shown on the client such as "Invalid username/password", etc. It currently may display raw server errors depending on the circumstance.

This was broken when I refactored the store/service/controller infrastructure.

The main controller (or service? figure it out) should make sure that only approved error messages are being sent back to the client for use on the homepage.

### PR DESCRIPTION
**Pull request by davidvorona**
Closes #302

- replace client-friendly stack trace with generalized 'internal server error'
- better way to handle specific errors that we need to send back to the client

### CODE REVIEW

File: server/service/user/auth.service.js [hunk 1]
```diff
@@ -26,6 +26,22 @@ const generateSecret = () => {
 };
 
 const authService = {
+    /**
+     * Result returned by auth service methods that have special-case errors.
+     *
+     * @typedef {Object} authResult
```
Comment (evanw555) [inline server/service/user/auth.service.js]: I might just be anal (:hankey:) but I feel like `authResult` as a type should be capitalized as `AuthResult`

File: server/service/user/auth.service.js [hunk 2]
```diff
@@ -36,7 +52,7 @@ const authService = {
             const userID = await usersStore.addUser(client, username);
             
             if (userID === null) {
-                throw new Error('Username already exists.');
+                return { err: 'Username already exists.' };
```
Comment (evanw555) [inline server/service/user/auth.service.js]: Return an actual error message and use `err.message` to extract the data from it. If you do this, make sure to change the `authResult` type definition too

File: server/service/user/auth.service.js [hunk 3]
```diff
@@ -65,28 +81,34 @@ const authService = {
...
+     * @returns {Promise<authResult>}
+     */
     authenticateUser: async (username, password) => {
         const client = await store.beginTransaction();
         try {
             const userID = await usersStore.getUserID(client, username);
 
             // fetch user's password components
-            const { salt, passhash } = await authStore.getPasswordComponents(null, userID);
+            const { salt, passhash } = await authStore.getPasswordComponents(client, userID);
```
Comment (evanw555) [inline server/service/user/auth.service.js]: Nice

File: server/service/user/auth.service.js [hunk 4]
```diff
@@ -26,6 +26,22 @@ const generateSecret = () => {
...
+     * @property {*} [token]
+     * @property {boolean} [status]
+     */
+
+    /**
+     *
+     * @param username
+     * @param password
+     * @param email
+     * @returns {Promise<authResult>}
```
[the remark this answers is not in the document]
Comment (davidvorona) [reply server/service/user/auth.service.js]: Doesn't break but it doesn't like it

File: server/service/user/auth.service.js [hunk 5]
```diff
@@ -26,6 +26,22 @@ const generateSecret = () => {
 };
 
 const authService = {
+    /**
+     * Result returned by auth service methods that have special-case errors.
+     *
+     * @typedef {Object} authResult
+     * @property {string} [err]
+     * @property {*} [token]
+     * @property {boolean} [status]
```
[the remark this answers is not in the document]
Comment (evanw555) [reply server/service/user/auth.service.js]: nvm it's used for the token methods

[the remark this answers is not in the document]
Comment (davidvorona) [reply server/service/user/auth.service.js]: it does in `removeToken`, didn't want to make a separate one though I could

**Pull request merged**